release.md

```diff
@@ -0,0 +1,73 @@
+- A hotfix, where the PATCH is bumped (`0.{x}.{y} -> 0.{x}.{y+1}`)
+
+Only releases are tracked and tagged in the `master` branch, while development is done in the `develop` branch. To signal this, the version in the `develop` branch should always be one MINOR ahead of `master` and follow the `MAJOR.{MINOR+1}devPATCH` format (e.g. if the latest release tagged in `master` was `0.4.3`, then the version in `develop` should be `0.5dev3`). Through this format, we mark which release is under development and how many patches have occured since its development started (which tells us how many times we brought in changes done directly in `master` through an hotfix).
+
+The version number is centralized in the `VERSION.txt` file and is shared between all the Pulser packages.
+
+
+## Preparing a scheduled release
+
+A scheduled release is the result of a series of features that were added to the `develop` branch over time. The release process starts out with the creation of a release branch, which should be branched out from `develop` to contain all the desired features and be named `release/v{x}.{y}.{z}`, where `x, y, z` are the MAJOR, MINOR and PATCH of the version to be released (though usually a scheduled release will have PATCH=0). 
+
+In the release branch, no other features can be added. Changes to the documentation and bug fixes are allowed, but should only be done when the development in the `develop` branch needs to continue while the release is being prepared; otherwise, do all the changes in `develop` before checking out the release branch. Note that the release branch will ultimately be *merged* to the `master` branch *without being squashed*, so keep the ammount of commits small and document them well to preserve the quality of the history.
+
+Crucially, the release branch must feature a commit changing the development version in `VERSION.txt` to the desired version of the release. For a minor release, this should be of the form `{x}.{y}dev{z} -> {x}.{y}.0`. 
+
+Finally, open a PR from the `release/v{x}.{y}.{z}` branch to `master`, have someone review and accept the changes introduced in the `release` branch (all the changes done in `develop` will be there as well, but those have already been reviewed) and merge the branch to `master` **without squashing the commits**.
+
+
+## Preparing a hotfix
+
+Unlike with a scheduled release, a hotfix serves only to fix bugs found in the latest release. The hotfix branch must be branched out from `master` and feature only the changes required to fix any bugs.
+
+Along with the bug fixes, the hotfix branch must also have a commit updating the version with an increment of the PATCH, ie `{x}.{y}.{z} -> {x}.{y}.{z+1}`.
+
+When ready, open a PR to merge the hotfix branch to `master` and, once that is reviewed and accepted, **squash and merge the commits** (note the difference with respect to the scheduled release procedure).
```

Comment:
From my understanding, this squash rule is new (just like the naming convention), because v0.5.1 did not apply it (0731ae4, 9304b65 and 5bf2d57 all appear in the master's commit history).

Reply:
There was no explicit rule before, so it is true you will find violations of these rules in the past history.
That being said, these commits don't all point to the squash rule you're referring to:
- [0731ae4](https://github.com/pasqal-io/Pulser/commit/0731ae486fe7056ce7be344fbb811d100b0e47f9) : This is an hotfix + dev version bump which was squashed and merged back to develop. Following these rules, it would still appear in the `master` history, with two differences: 1) the version would be `0.6dev1`; 2) it would not be squashed, so the hotfix changes would appear in a separate commit from the version bump
- [9304b65](https://github.com/pasqal-io/Pulser/commit/9304b651587568044076241948d17993e7364355) & [5bf2d57](https://github.com/pasqal-io/Pulser/commit/5bf2d57eb8692adc16a13a1c6f4b962705b6fb61) : These refer to what you mention, the version bump was done separately from the hotfix, while now it would be done together in the same squashed commit